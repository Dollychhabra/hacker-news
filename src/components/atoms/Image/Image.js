import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  alt: PropTypes.string.isRequired,
  srcset: PropTypes.string,
  src: PropTypes.string.isRequired,
  className: PropTypes.string,
};

const Image = ({ className, src, srcset, sizes, alt, ...other }) => {
  function addErrorClass(e) {
    e.target.style.display = 'none';
  }

  return (
    <React.Fragment>
      <img
        className={className}
        src={src}
        srcSet={srcset || null}
        alt={alt}
        onError={addErrorClass}
        {...other}
      />
    </React.Fragment>
  );
};

Image.defaultProps = {
  srcset: '',
  className: '',
};

Image.propTypes = propTypes;

export default Image;
