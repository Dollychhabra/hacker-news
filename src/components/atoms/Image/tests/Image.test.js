import React from 'react';
import { shallow } from 'enzyme';
import Image from '../Image';

describe('Image Component', () => {
  const props = {
    alt: 'aa',
    src: '/',
    isDynamic: true,
  };

  test('should render correctly and match snapshot', () => {
    const component = shallow(<Image {...props} />);
    expect(component).toMatchSnapshot();
  });

});
