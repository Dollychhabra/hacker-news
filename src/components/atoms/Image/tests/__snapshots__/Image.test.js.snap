// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Component should render correctly and match snapshot 1`] = `ShallowWrapper {}`;
